packet.go

```diff
@@ -56,6 +56,11 @@ func marshalFileInfo(b []byte, fi os.FileInfo) []byte {
 	flags, fileStat := fileStatFromInfo(fi)
 
 	b = marshalUint32(b, flags)
+
+	return marshalFileStat(b, flags, fileStat)
+}
+
+func marshalFileStat(b []byte, flags uint32, fileStat *FileStat) []byte {
```

Comment:
Spin this out so that we can call it elsewhere, like if an open/(f)setstat packet has a `*FileStat` as it’s `p.Attrs`.